### PR TITLE
Use WKWebView for authorization

### PR DESCRIPTION
**Pull request by jlawton**
### Issue Link :link:
Use of UIWebView has become a problem when submitting to the Apple App Store. Mentioned in #669 and #647. This is an alternative to #710.

### Goals :soccer:
* Replace UIWebView with WKWebView for authorization.

### Implementation Details :construction:
* Swap in WKWebView.
* Update delegate methods to support WKWebView.
* Remove use of URLSession to load some requests, which was required when using UIWebView, but should no longer be.
* Update Xcode projects according to Xcode's automatic fix suggestions.

### Testing Details :mag:
* Tested by running the Sample App and logging into Box, using the updated SDK.

### CODE REVIEW

**Comment by vikrambatra08** [comment]
May I know when this pull request will get merged. As I am blocked to publish my build to AppStore, could you please do a merge. Request you to please let me know if we have any alternative for this.
Thanks :)

**Comment by PJSimon** [comment]
@vikrambatra08 I understand this is a blocker for you, so I'll get to it later today! Sorry for the delay!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -439,34 +374,28 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
...
-        BOXLog(@"Was not authenticated, launching URLSession and not loading the request in the web view");
-        [self.activityIndicator startAnimating];
-        NSURLSessionDataTask *dataTask = [self.URLSession dataTaskWithRequest:request];
-        [dataTask resume];
-        BOXLog(@"URLSessionDataTask is %@", dataTask);
-        return NO;
+        decisionHandler(WKNavigationActionPolicyCancel);
+        return;
+    }
+    if (@available(iOS 11.0, *)) {
```
Comment (PJSimon) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:380]: Does this `if` need an `else`?

Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:380]: I think the else is just to fall through.

**Review by PJSimon — APPROVED** [review]

**Comment by vikrambatra08** [comment]
Thank you so much for merging this PR @PJSimon !. But I can still find UIWebView matches in following files.

Binary file ./Carthage/Build/iOS/BoxContentSDKTestFramework.framework.dSYM/Contents/Resources/DWARF/BoxContentSDKTestFramework matches

Binary file ./Carthage/Build/iOS/BoxContentSDK.framework.dSYM/Contents/Resources/DWARF/BoxContentSDK matches

Seems this is happening only with Carthage, do we have CocoaPod available for this release? If not, as with Carthage we are getting dSYM file (mentioned above) created, can we delete those? 
Can you please help me on this as well?

**Pull request merged**